### PR TITLE
Add bucket ls

### PR DESCRIPTION
**Pull request by andriytk**

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -296,7 +296,14 @@ namespace rgw::sal {
     bufferlist& blr = bl;
     auto iter =blr.cbegin();
     mbinfo.decode(iter); //Decode into MotrBucketInfo.
+
     info = mbinfo.info;
+    ldpp_dout(dpp, 0) << "get_bucket_info(): bucket id  = " << info.bucket.bucket_id << dendl;
+    rgw_placement_rule placement_rule;
+    placement_rule.name = "default";
+    placement_rule.storage_class = "STANDARD";
+    info.placement_rule = placement_rule;
```
Comment (andriytk) [inline src/rgw/rgw_sal_motr.cc:305]: Why is this needed? Shouldn't it be there already in the decoded info?

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -1356,17 +1505,22 @@ namespace rgw::sal {
     } else {
       placement_rule.name = "default";
       placement_rule.storage_class = "STANDARD";
-      info.placement_rule = placement_rule;
-      info.bucket = b;
-      bucket = std::make_unique<MotrBucket>(this, info, u);
-      *existed = false;
+      bucket = std::make_unique<MotrBucket>(this, b, u);
```
Comment (andriytk) [inline src/rgw/rgw_sal_motr.cc:1508]: Don't you want to store the info with the placement_rule? Do you think it's better to just return it on bucket get? I'm fine with both ways, just curious.

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
Comment (siningwuseagate) [reply src/rgw/rgw_sal_motr.cc:305]: I tried your put patch, it had some issue about the placement, so I hand coded it for my debugging. I should work without that after the latest fix. Let me confirm it.

Comment (andriytk) [reply src/rgw/rgw_sal_motr.cc:305]: Well, we can do it this way if we are not storing it.

File: src/rgw/rgw_sal_motr.cc (hunk 2 above)
Comment (siningwuseagate) [reply src/rgw/rgw_sal_motr.cc:1508]: The placement is set and stored in later lines together with other bucket info members.
see line 1517 - 1521
